### PR TITLE
Add WEARABLES_COMPILE_PYTHON_BINDINGS flag

### PR DESCRIPTION
**Pull request by lrapetti**
Following https://github.com/robotology/wearables/pull/162, we should enable the `WEARABLES_COMPILE_PYTHON_BINDINGS` when `ROBOTOLOGY_USES_PYTHON` is enabled

### CODE REVIEW

**Comment by lrapetti** [comment]
@traversaro @GiulioRomualdi I am not sure if I am missing something else to be enabled or other cases to be tested

**Review by traversaro — APPROVED** [review]

**Comment by traversaro** [comment]
Thanks, I also added Python dependencies to conda so that conda packages now are correctly built.

File: cmake/Buildwearables.cmake [hunk 1]
```diff
@@ -23,3 +25,10 @@ ycm_ep_helper(wearables TYPE GIT
                       iDynTree
                       robometry
               CMAKE_ARGS ${WEARABLES_CMAKE_ARGS})
+
+if(ROBOTOLOGY_USES_PYTHON)
+  list(APPEND wearables_CONDA_DEPENDENCIES python)
+  list(APPEND wearables_CONDA_DEPENDENCIES pybind11)
+  # https://conda-forge.org/docs/maintainer/knowledge_base.html#pybind11-abi-constraints
+  list(APPEND bipedal-locomotion-framework_CONDA_DEPENDENCIES pybind11-abi)
```
Comment (RiccardoGrieco) [inline cmake/Buildwearables.cmake:33]:
```suggestion
  list(APPEND wearables_CONDA_DEPENDENCIES pybind11-abi)
```
I suppose this is a copy-paste typo.

Comment (lrapetti) [reply cmake/Buildwearables.cmake]: Thanks for noticing!

Comment (traversaro) [reply cmake/Buildwearables.cmake]: Yes, thanks!

**Comment by traversaro** [comment]
Conda is failing with:
~~~
2022-09-14T11:00:31.5259665Z FAILED: bindings/python/CMakeFiles/pybind11_wearables.dir/wearables.cpp.o 
2022-09-14T11:00:31.5262508Z /usr/share/miniconda3/envs/test/bin/x86_64-conda-linux-gnu-c++ -DUSING_DEPRECATED_UPPERCASE_YARP_OS_TARGET -Dpybind11_wearables_EXPORTS -I/home/runner/work/robotology-superbuild/robotology-superbuild/src/wearables/bindings/python/msgs/include -I/home/runner/work/robotology-superbuild/robotology-superbuild/build/src/wearables/msgs -isystem /usr/include/python3.8 -isystem /home/runner/work/robotology-superbuild/robotology-superbuild/build/install/include -fvisibility-inlines-hidden -std=c++17 -fmessage-length=0 -march=nocona -mtune=haswell -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O2 -ffunction-sections -pipe -isystem /usr/share/miniconda3/envs/test/include -O3 -DNDEBUG -fPIC -fvisibility=hidden -flto -fno-fat-lto-objects -std=gnu++17 -MD -MT bindings/python/CMakeFiles/pybind11_wearables.dir/wearables.cpp.o -MF bindings/python/CMakeFiles/pybind11_wearables.dir/wearables.cpp.o.d -o bindings/python/CMakeFiles/pybind11_wearables.dir/wearables.cpp.o -c /home/runner/work/robotology-superbuild/robotology-superbuild/src/wearables/bindings/python/wearables.cpp
2022-09-14T11:00:31.5264364Z In file included from /usr/include/python3.8/Python.h:8,
2022-09-14T11:00:31.5264702Z                  from /usr/share/miniconda3/envs/test/include/pybind11/detail/common.h:212,
2022-09-14T11:00:31.5265066Z                  from /usr/share/miniconda3/envs/test/include/pybind11/attr.h:13,
2022-09-14T11:00:31.5265421Z                  from /usr/share/miniconda3/envs/test/include/pybind11/detail/class.h:12,
2022-09-14T11:00:31.5265769Z                  from /usr/share/miniconda3/envs/test/include/pybind11/pybind11.h:13,
2022-09-14T11:00:31.5266368Z                  from /home/runner/work/robotology-superbuild/robotology-superbuild/src/wearables/bindings/python/wearables.cpp:8:
2022-09-14T11:00:31.5267003Z /usr/include/python3.8/pyconfig.h:3:12: fatal error: x86_64-linux-gnu/python3.8/pyconfig.h: No such file or directory
2022-09-14T11:00:31.5267454Z     3 | #  include <x86_64-linux-gnu/python3.8/pyconfig.h>
2022-09-14T11:00:31.5267707Z       |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-14T11:00:31.5268007Z compilation terminated.
~~~

It seems that for some reason wearables is finding the system Python instead of the conda one, let me check.

**Comment by traversaro** [comment]
Out of 12 `Found Python3:` occurences in the log, wearables is the only one wrong for some reason:
~~~
2022-09-14T10:59:13.7007084Z --  [x] Plugin: iwearactuators_wrapper (ENABLE_iwearactuators_wrapper)
2022-09-14T10:59:13.7007543Z --  [x] Plugin: ixsensmvncontrol_wrapper (ENABLE_ixsensmvncontrol_wrapper)
2022-09-14T10:59:13.7008224Z -- Found Python3: /usr/bin/python3.8 (found suitable version "3.8.10", minimum required is "3.6") found components: Interpreter Development Development.Module Development.Embed 
2022-09-14T10:59:13.7008890Z -- Performing Test HAS_FLTO
2022-09-14T10:59:13.7009206Z -- Performing Test HAS_FLTO - Success
2022-09-14T10:59:13.7009620Z -- Found pybind11: /usr/share/miniconda3/envs/test/include (found version "2.10.0")
2022-09-14T10:59:13.7010058Z -- Added files for bindings named MsgsBindings in Wearables.
~~~

**Comment by traversaro** [comment]
I did a local test, and the problem seems to be the outdated vendored `FindPython3.cmake` module vendored in wearables, added in https://github.com/robotology/wearables/pull/162/files . FindPython3 was added in CMake 3.12 (see https://cmake.org/cmake/help/latest/module/FindPython3.html), probably we can just bump the minimum cmake required in wearables to 3.12 (or even 3.16, the superbuild requires 3.16), and remove the vendored FindPython3 ? @GiulioRomualdi @lrapetti

**Comment by traversaro** [comment]
I bumped wearables version to 1.6.0 as it should include https://github.com/robotology/wearables/pull/165 .

**Review by RiccardoGrieco — APPROVED** [review]

**Comment by traversaro** [comment]
The failures are due to https://github.com/robotology/robotology-superbuild/issues/1271, https://github.com/robotology/robotology-superbuild/issues/1269 and https://github.com/robotology/robotology-superbuild/issues/1266 , so we can merge.

**Pull request merged**